### PR TITLE
Converted TodoCount to ES2015

### PR DESCRIPTION
**Pull request by nii236**
Mmm ES2015

### CODE REVIEW

**Comment by nii236** [comment]
Handy info:

http://www.newmediacampaigns.com/blog/refactoring-react-components-to-es6-classes

File: src/components/TodoCount.jsx [hunk 1]
```diff
@@ -1,14 +1,7 @@
...
-export default React.createClass({
-  displayName: 'TodoCount',
-
-  propTypes: {
-    count: React.PropTypes.number,
-    completedCount: React.PropTypes.number
-  },
-
-  render: function () {
+export default class TodoCount extends React.Component {
```
Comment (pzi) [inline src/components/TodoCount.jsx]: Does this component actually work as intended?

Check @levibuzolic's example: https://github.com/pzi/simple-todo-webapp/blob/master/src/components/examples/ES6StandardComponent.jsx#L41-L43

**Comment by nii236** [comment]
Its funny that you ask that! The original branch didn't work in the first place. The `completedCount` was not updating when I check one of the items.

I can confirm that the functionality is unchanged between my PR and the master branch.

File: src/components/TodoCount.jsx [hunk 2]
```diff
@@ -18,4 +11,11 @@ export default React.createClass({
...
   }
-});
+}
+
+TodoCount.propTypes = {
+  count: React.PropTypes.number,
+  completedCount: React.PropTypes.number
+};
+
+export default TodoCount;
```
Comment (pzi) [inline src/components/TodoCount.jsx]: Now you are exporting 2x my friend :)

Comment (nii236) [inline src/components/TodoCount.jsx]: Ahhhhhh!

**Comment by levibuzolic** [comment]
When using `export default` I think you can actually get away with exporting at the top, but specifying `propTypes` at the bottom. Because babel will actually transform it to have the `module.exports = ` at he very bottom of the file. See: https://goo.gl/ykGOkF

But it's probably easier just to use the new syntax:

```javascript
import React from 'react';

export default class TodoCount extends React.Component {
  static propTypes = {
    count: React.PropTypes.number,
    completedCount: React.PropTypes.number
  }

  render() {
    return (
      <dl>
        <dt>Number of Todos:</dt>
        <dd>{ this.props.count }</dd>
        <dt>Completed Todos:</dt>
        <dd>{ this.props.completedCount }</dd>
      </dl>
    );
  }
}

**Comment by levibuzolic** [comment]
Also the reason why the count doesn't get updated is because you're updating state in `TodoItems` while the count renderer is in `TodoApp` (one level above) so it has no knowledge of the new state. Also points out that you have the same state in multiple locations.

**Comment by pzi** [comment]
Yeah that would've happened when I moved TodoItems into a separate file and did a half-assed job :(

**Comment by pzi** [comment]
And yeah, I specifically chose to write all my components in ES5 style because it makes it easier form me to learn as there are heaps of resources out there in that style. Once I feel comfortable with what I am actually doing I will then move up to ES6 style

**Pull request closed without merging**